sympy/functions/elementary/exponential.py

```diff
@@ -964,9 +970,11 @@ def eval(cls, x, k=None):
                 return S.Exp1
             if x is S.Infinity:
                 return S.Infinity
+            if x.is_zero:
+                return S.Zero
 
         if fuzzy_not(k.is_zero):
-            if x is S.Zero:
+            if x is S.Zero or x.is_zero:
```

Comment:
Is it beneficial to so if `x is S.Zero or x.is_zero` rather than just `if x.is_zero`? In the case that x is `S.Zero`, both essentially boil down to an attribute lookup:
```julia
In [5]: x = S.Zero                                                                                                                

In [6]: %timeit bool(x is S.Zero or x.is_zero)                                                                                    
212 ns ± 1.97 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [7]: %timeit bool(x.is_zero)                                                                                                   
184 ns ± 5.24 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
In any other case the `x is S.Zero` check will be False and we still end up calling `x.is_zero`.

Even if this was faster I would be unsure about this change: it would be better to optimise `S.Zero.is_zero` than pepper things like this around the codebase.